src/model_io/codecs/src/SDFormatDocument.cpp

```diff
@@ -0,0 +1,571 @@
+// SPDX-FileCopyrightText: Fondazione Istituto Italiano di Tecnologia (IIT)
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "SDFormatDocument.h"
+
+#include <iDynTree/AccelerometerSensor.h>
+#include <iDynTree/FixedJoint.h>
+#include <iDynTree/GyroscopeSensor.h>
+#include <iDynTree/Link.h>
+#include <iDynTree/Model.h>
+#include <iDynTree/ModelTransformers.h>
+#include <iDynTree/PrismaticJoint.h>
+#include <iDynTree/RevoluteJoint.h>
+#include <iDynTree/SolidShapes.h>
+#include <iDynTree/Utils.h>
+
+#ifdef IDYNTREE_USES_SDFORMAT
+#include <gz/math/Inertial.hh>
+#include <gz/math/Pose3.hh>
+#include <gz/math/Vector3.hh>
+#include <sdf/sdf.hh>
+#endif
+
+#include <algorithm>
+#include <string>
+
+namespace iDynTree
+{
+
+#ifdef IDYNTREE_USES_SDFORMAT
+    namespace
+    {
+
+        // Helper function to convert gz::math::Pose3d to iDynTree::Transform
+        iDynTree::Transform poseToTransform(const gz::math::Pose3d &pose)
+        {
+            iDynTree::Transform transform;
+
+            // Convert rotation from quaternion to rotation matrix
+            const gz::math::Quaterniond &quat = pose.Rot();
+            iDynTree::Rotation rotation;
+
+            double w = quat.W(), x = quat.X(), y = quat.Y(), z = quat.Z();
+
+            rotation(0, 0) = 1 - 2 * (y * y + z * z);
+            rotation(0, 1) = 2 * (x * y - w * z);
+            rotation(0, 2) = 2 * (x * z + w * y);
+
+            rotation(1, 0) = 2 * (x * y + w * z);
+            rotation(1, 1) = 1 - 2 * (x * x + z * z);
+            rotation(1, 2) = 2 * (y * z - w * x);
+
+            rotation(2, 0) = 2 * (x * z - w * y);
+            rotation(2, 1) = 2 * (y * z + w * x);
+            rotation(2, 2) = 1 - 2 * (x * x + y * y);
```

Comment:
I am a bit afraid of this untested logic. Can you use either a gz-math helper or iDynTree helper to convert from quaternion to rotation?